cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -372,26 +379,77 @@ impl ClientTrait for CatalogClient {
     }
 }
 
+/// Collects a stream of search results into a container, returning the total count as well.
+///
+/// Note: it is assumed that the first element of the stream contains the total count.
+async fn collect_search_results<T, E>(
+    stream: impl Stream<Item = Result<StreamItem<T>, E>>,
+    limit: SearchLimit,
+) -> Result<(ResultCount, Vec<T>), E> {
+    let mut count = None;
+    let actual_limit = if let Some(checked_limit) = limit {
+        checked_limit.get() as usize
+    } else {
+        // If we survive long enough that this becomes a problem, I'll fix it
```

Comment:
suggestion blocking: add your phone number to this comment

Reply:
1-800-NO